builder/haddock-builder.nix

```diff
@@ -26,25 +29,17 @@ let
 
   fullName = "${componentDrv.name}-haddock";
 
+  # These config files are like the one used in the build derivation,
+  # but `chooseDrv` will be used to map all the references to libraries
+  # to their haddock derivation.
   docsConfigFiles = makeConfigFiles {
     inherit (package) identifier;
     inherit component fullName flags needsProfiling;
     chooseDrv = p: p.haddock;
   };
 
-  finalConfigureFlags = lib.concatStringsSep " " (
-    [ "--prefix=${componentDrv}"
-      "--docdir=${docdir "$doc"}"
-      "${haskellLib.componentTarget componentId}"
-      "$(cat ${docsConfigFiles}/configure-flags)"
-      # GHC
-      "--with-ghc=${ghc.targetPrefix}ghc"
-      "--with-ghc-pkg=${ghc.targetPrefix}ghc-pkg"
-      "--with-hsc2hs=${ghc.targetPrefix}hsc2hs"
-    ]
-      ++ configureFlags
-      ++ (ghc.extraConfigureFlags or [])
-    );
+  finalConfigureFlags = buildConfigureFlags
+    + " --docdir=${docdir "$doc"} $(cat ${docsConfigFiles}/configure-flags)";
```

Comment:
Do later options override earlier ones?

Reply:
Just in case I have refactored this so that commonConfigureFlags is passed thought so that there should be no overlap.